### PR TITLE
Improve response handling for http

### PR DESCRIPTION
**Pull request by nadheesh**
## Purpose
resolves https://github.com/wso2-enterprise/choreo/issues/24026
https://github.com/wso2-enterprise/choreo/issues/23947

### CODE REVIEW

File: ballerina/toolkit.bal [hunk 1]
```diff
@@ -325,3 +321,52 @@ isolated function extractPathParams(string path, ParameterSchema? pathParameters
...
+            return {
+                code: response.statusCode,
+                contentType: mime:IMAGE_PNG,
+                payload: check response.getBinaryPayload()
+            };
+        }
+        _ => {
+            return {
+                code: response.statusCode,
+                contentType: mime:IMAGE_PNG,
```
Comment (JayaniH) [inline ballerina/toolkit.bal:365]: is the content type correct for the default case?

File: ballerina/toolkit.bal [hunk 2]
```diff
@@ -78,8 +79,10 @@ type HttpInput record {|
 public type HttpOutput record {|
     # HTTP status code of the response
     int code;
+    # Content type
+    string contentType;
     # Content of the response
-    string payload?;
+    json|xml payload?;
```
Comment (JayaniH) [inline ballerina/toolkit.bal:85]: does this capture the other types also?

Comment (nadheesh) [reply ballerina/toolkit.bal:85]: json captures all we are expecting expect xml

File: ballerina/toolkit.bal (hunk 1 above)
Comment (nadheesh) [reply ballerina/toolkit.bal]: fixed

**Review by JayaniH — APPROVED** [review]

**Pull request merged**